File: tests/benchmark/corpus/python/safe/safe_data_exfil_user_input_echo.py
import requests
from flask import Flask, request

app = Flask(__name__)


# DATA_EXFIL safe: plain user input echoed into a fixed-destination
# requests.post body must not fire. Sensitivity-gate strips the cap
# because the source is Plain-tier (raw user input).
@app.route('/forward', methods=['POST'])
def forward():
    message = request.form.get('message')
    requests.post('https://telemetry.internal/forward', json={'message': message})
    return 'ok'

File: tests/benchmark/corpus/ruby/safe/safe_data_exfil_user_input_echo.rb
require 'rest-client'

# DATA_EXFIL safe: plain user input echoed into a RestClient.post body
# at a fixed destination URL must not fire. Sensitivity-gate strips the
# cap for Plain-tier sources.
def forward_message(params)
  message = params[:message]
  RestClient.post(
    'https://telemetry.internal/forward',
    { message: message }.to_json
  )
end
